modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/ClientHandler.java

```diff
@@ -203,6 +215,16 @@ public ClientHandler(
 
         // set the latest openConnections map to MBean data during connection creation
         metrics.setConnectionsPerHosts(openConnections);
+
+        //load properties related to message size validation
+        isMessageSizeValidationEnabled = cfg.getMessageSizeValidationEnabled();
+        try {
```

Comment:
Reading the max message size is required only if isMessageSizeValidationEnabled is true. Shouldn't we put this try block inside a if(isMessageSizeValidationEnabled)?